src/converse-notification.js

```diff
@@ -32,32 +32,28 @@ converse.plugins.add('converse-notification', {
             play_sounds: true,
             sounds_path: api.settings.get("assets_path")+'/sounds/',
             notification_icon: 'logo/conversejs-filled.svg',
-            notification_delay: 5000
+            notification_delay: 5000,
+            notify_nicknames_without_references: false
         });
 
         _converse.shouldNotifyOfGroupMessage = function (message) {
             /* Is this a group message worthy of notification?
              */
-            let notify_all = api.settings.get('notify_all_room_messages');
-            const jid = message.getAttribute('from'),
-                resource = Strophe.getResourceFromJid(jid),
-                room_jid = Strophe.getBareJidFromJid(jid),
-                sender = resource && Strophe.unescapeNode(resource) || '';
-            if (sender === '' || message.querySelectorAll('delay').length > 0) {
-                return false;
-            }
-            const room = _converse.chatboxes.get(room_jid);
-            const body = message.querySelector('body');
-            if (body === null) {
-                return false;
-            }
-            const mentioned = (new RegExp(`\\b${room.get('nick')}\\b`)).test(body.textContent);
-            notify_all = notify_all === true ||
-                (Array.isArray(notify_all) && notify_all.includes(room_jid));
-            if (sender === room.get('nick') || (!notify_all && !mentioned)) {
-                return false;
+            const jid = message.getAttribute('from');
+            const room_jid = Strophe.getBareJidFromJid(jid);
+            const notify_all = api.settings.get('notify_all_room_messages');
+            let is_mentioned = false;
+            
+            if (api.settings.get('notify_nicknames_without_references')) {
+                const room = _converse.chatboxes.get(room_jid);
+                const body = message.querySelector('body');
+                is_mentioned = (new RegExp(`\\b${room.get('nick')}\\b`)).test(body.textContent);
             }
-            return true;
+            
+            const result = notify_all === true
+                || (Array.isArray(notify_all) && notify_all.includes(room_jid))
+                || is_mentioned;
+            return !!result;
```

Comment:
Looks like you removed this line:
```
            if (sender === room.get('nick') || (!notify_all && !mentioned)) {	
                return false;	
            }
```

which means that the user will also be notified for messages he sends himself, which is not what we want.
This would have been easily caught with a test if we had one for it. Please update the tests to check that users aren't notified for messages they send themselves.

I think you can just put this line near the beginning of the method to fix this:
```
            if (sender === room.get('nick')) {
                return false;	
            }
```